src/jit/utils.h

```diff
@@ -82,40 +82,70 @@ class JitSimplerHashBehavior
     }
 };
 
-#ifdef DEBUG
-/**************************************************************************/
+#if defined(DEBUG) || defined(INLINE_DATA)
 
-/* to be used as static variables - no constructors/destructors, assumes zero 
-   initialized memory */
+// ConfigMethodRange describes a set of methods, specified via their
+// hash codes. This can be used for binary search and/or specifying an
+// explicit method set.
+//
+// Note method hash codes are not necessarily unique. For instance
+// many IL stubs may have the same hash.
+//
+// If range string is null or just whitespace, range includes all
+// methods.
+//
+// Parses values as decimal numbers.
+//
+// Examples:
+//
+//  [string with just spaces] : all methods
+//                   12345678 : a single method
+//          12345678-23456789 : a range of methods
+// 99998888 12345678-23456789 : a range of methods plus a single method
 
 class ConfigMethodRange
 {
 
 public:
-    bool contains(class ICorJitInfo* info, CORINFO_METHOD_HANDLE method);
 
-    inline void ensureInit(const wchar_t* rangeStr)
+    // Default capacity. Number of ranges is half this.
+    enum
+    {
+        DEFAULT_CAPACITY = 100
+    };
+
+    // Does the range include this method's hash?
+    bool Contains(class ICorJitInfo* info, CORINFO_METHOD_HANDLE method);
+
+    // Ensure the range string has been parsed.
+    void EnsureInit(const wchar_t* rangeStr, unsigned capacity = DEFAULT_CAPACITY)
     {
-        // make sure that the memory was zero initialized
+        // Make sure that the memory was zero initialized
         assert(m_inited == 0 || m_inited == 1);
 
         if (!m_inited)
         {
-            initRanges(rangeStr);
+            InitRanges(rangeStr, capacity);
             assert(m_inited == 1);
         }
     }
 
-private:
-    void initRanges(__in_z LPCWSTR rangeStr);
+    // Error checks
+    bool Error()          const { return m_badChar != 0; }
+    size_t BadCharIndex() const { return m_badChar - 1; }
 
 private:
-    unsigned char m_lastRange;                   // count of low-high pairs
-    unsigned char m_inited;
-    unsigned m_ranges[100];                      // ranges of functions to Jit (low, high pairs).  
+
+    void InitRanges(const wchar_t* rangeStr, unsigned capacity);
+
+    unsigned  m_entries;    // number of entries in the range array
+    unsigned  m_lastRange;  // count of low-high pairs
+    unsigned  m_inited;     // 1 if range string has been parsed
+    size_t    m_badChar;    // index + 1 of any bad character in range string
+    unsigned* m_ranges;     // ranges of functions to Jit (low, high pairs)
 };
```

Comment:
Seems weird that we don't have:

struct Range { unsigned low, high; }
Range* m_ranges;

not terribly important, I guess...